src-java/base-topology/base-messaging/src/main/java/org/openkilda/messaging/validation/JsonIncludeObjectHavingNonNullFieldOrNonEmptyList.java

```diff
@@ -0,0 +1,65 @@
+/* Copyright 2023 Telstra Open Source
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License");
+ *   you may not use this file except in compliance with the License.
+ *   You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ */
+
+package org.openkilda.messaging.validation;
+
+import java.lang.reflect.Field;
+import java.lang.reflect.Modifier;
+import java.util.List;
+
+/**
+ * This class is a custom Jackson's JsonInclude validation intended to exclude from JSON
+ * an object containing only null fields and non-null but empty Lists.
+ */
+public final class JsonIncludeObjectHavingNonNullFieldOrNonEmptyList {
+
+    @Override
+    public boolean equals(Object obj) {
+        if (obj == null) {
+            return true;
+        }
+
+        try {
+            for (Field field : obj.getClass().getDeclaredFields()) {
+                if (Modifier.isTransient(field.getModifiers())) {
+                    continue;
+                }
+                if (field.getType().isPrimitive()) {
+                    return false;
+                }
+
+                field.setAccessible(true);
+                if (List.class.isAssignableFrom(field.getType())) {
+                    if (!((List<?>) field.get(obj)).isEmpty()) {
+                        return false;
+                    }
+                } else {
+                    if (field.get(obj) != null) {
+                        return false;
+                    }
+                }
+            }
+        } catch (IllegalAccessException e) {
+            throw new IllegalStateException("Cannot determine whether to include a field because cannot access it", e);
```

Comment:
I'm wondering if we should fail the response in case we have problem filtering out some fields just for convenience. I mean, for me this doesn't sound like ERROR. May be warning in logs would be enough?

Reply:
I considered it, but then decided to leave it like it is. This exception means that the logic of include/exclude is not compatible with the given model class. Meaning either change the class, or implement different logic for include/exclude.

Reply:
I changed the type and the message. Should be more clear now.